### ISSUE TITLE
2.5.0 Beta

### ISSUE THREAD

**Issue by mackworth** [dscottbuch/cTiVo#197]
A place for discussion & comments on newest beta.

**Comment by swcoop1** [comment]
I have just upgraded to 2.5.0 beta, and am getting "encode failed" errors in the log.  Currently downloading shows in "Decrypted TiVo Show" format, which so far is working.

I also average about 700KB/sec, despite restarts of computer and TiVo, whether wired or wireless from TiVo and computer.

Please let me know if I can provide any other info.  Thanks!

[log.docx](https://github.com/dscottbuch/cTiVo/files/477969/log.docx)

**Comment by mackworth** [comment]
Hi, can you set the logging level to Major and redo the first show? (Which Format?)  Also, are you only getting 700KB with Decrypted?  If so, then during the download, can you set the logging to Verbose for about 10 seconds and back to Major? I've been working on a bottleneck, and you may have run into it.

**Comment by tannebil** [comment]
Installed it yesterday. Everything seems to work so far. Preferences/Channels showed a couple of MPEG4 channels (TBDHDP, UNVSOHD) on Comcast in my area (Pierce County, Washington) so I tried out the ffmpeg Comskip/5.1 format. No problem playing the result via an ATV4 and on my iPhone (have not checked 5.1 sound yet). As you'd expect, video quality was excellent and the process was fast, but the file is about 70% larger. 

Commercial marking missed the break between the last commercial block and the last program segment. The show was Full Frontal with Samantha Bee/Super Lobbyist. I'll need more shows to know if this was a one-off specific to that program or a bug.

Commercial skipping has alway worked well in iTunes and on my ATV3/4 but has been very shaky on iOS (better on iPad than iPhone). At least based on this one show, it still good on iTunes/ATV3/ATV4 but it better on my iPhone (but not my iPad). I've always assumed it was an iOS thing but I need more time watching shows I encoded on the past to be more confident that there's any real improvement.
.

**Comment by mackworth** [comment]
Welcome back!  

Have you tried the "test all channels" in Prefs>Channels? It will check all your channels at once(that you have  show on).

Good tip on the size of the file. If you'd like to try modifying that, just add `-crf xx` to the video options  "where 0 is lossless, 23 is default, and 51 is worst possible". They say that +6 doubles the size of the file, so if you want to try it back at about the same size, you could try 19. Maybe should tune that down for the default.

Have you seen any slowness on the download process as reported above?

Not much we can affect on iphone skipping. When you say "shaky", do you mean that it's not appearing at all, or that the marked values are wrong?

Everything else still showing up proper values in iTunes etc?

**Comment by tannebil** [comment]
No problems with the "test all channels" although I recall being a bit confused that clicking on the button to clear all test results didn't actually appear to change anything on the screen. Is there any way I'll know a channel switched to MPEG4 other than rerunning the "test all"?

I'll try the change quality parameter but since that's going to re-encode the file, I'm not sure what the ffmpeg format is offering me over my existing use of Handbrake encodes. I'll do some tests to see if there's a big performance or quality difference.

The downloads run at 4MB/sec according to cTiVo. I don't have past numbers but it feels about the same. My guess is that's pretty close to the max for my network connection (there is a Powerline Ethernet segment in it).

"Shaky" means unreliable. For example, instead of skipping to the next marker, it might skip to the last one but if i skip back from that point, it finds the intervening markers. That makes me think it's more likely to be an iOS problem rather than a cTiVo problem. I'll have a better feel for it now that both my phone and iPad are on 10. ATV lost one press skip in the new release so you have to pull down the list of chapter stops and select the one desired. Most likely an oversight or bug I figure.

Everything looks fine in the metadata although I noticed two issues in the latest iTunes unrelated to cTiVo. First, the season is now all caps, e.g. "SEASON 2" rather than "Season 2". Second, if I obtain a show from other sources that lacks the proper metadata, it's always dumped into "Home Videos" so I'd have to retag it. That always resulted in the show being moved into the "TV Shows" file hierarchy in past versions but It no longer moves it.  That leaves it out of my Plex app since I didn't configure it to scan that folder. Typical iTunes update where a few new little bugs get introduced.

**Comment by mackworth** [comment]
Great feedback; maybe better wording is necessary.  The Clear All Tests simply removes the completed tests from your download Queue. You can delete individual (or select all) channel results with the "-" button on that screen. 

If a channel switches, then it will be marked as such in the Channel listing. (The next time cTiVo tries that channel, it will fail with the audio-only symptom, cTiVo then marks it as H.264 and retries.)

 I'm very concerned about how to explain all this in the notes without overwhelming people.

The older powerline was nominally 200mbps, but in reality more like 50 in a typical household, so yah, 4MB/sec = 32mpbs real data throughput = ~40mbps bandwidth.

**Comment by tannebil** [comment]
Both the H.264 channels that I've picked up in the channels scan are 1080i so they are getting encoded. I hand-timed a 4:16 program segment using ffmpeg/5.1 and the download took 40 seconds and the encode 7:21. That's on a 2012 Mac Mini 16GB, 1TB SSD, 2.6Ghz i7 quad core.

Is there any way to check on the TiVo for MPEG4 channels short of recording a show from all of them? 

What would be very cool would be a log summary produced for every download with the key metadata for each show, the format details, and performance details like time to download, time to encode, and time to run Comskip. That would make it much easier to see when something has changed. Doesn't even need to be part of cTiVo. An external program or even a Perl script would help quite a bit. Ideally something that was in CSV or a similar format would be great. That way I could build up a history in a spreadsheet.

**Comment by mackworth** [comment]
That's about what I'm seeing as well on performance. depends on the resolution you're looking for. With a 1080i input, 4.0G 1hr input, mencoder at 960x540 takes about 31 minutes to encode, and about 

So...that's what the Test All Channels is for (and there is no other way available AFAIK). It will submit a Test download for each channel that you happen to have a show recorded on.  It only downloads a few megabytes for each show, which is enough to tell it the format.  Then it logs whether it is MP2 or H.264 and you can see the results in the Edit Channels panel.  So, counting the minute that we have to let TiVo cool down between downloads, over a couple hours you can check all your channels.

Not a bad thought to have a separate chronological log, which doesn't get erased as the regular one does.  Performance metrics are a little harder, as some of these things are happening in parallel depending on the flow you've selected, and/or if other shows are going on simultaneously. I can do clock time and CPU time, but they may not correlate.

**Comment by tannebil** [comment]
Real benchmarking is much harder than I had in mind. I was thinking more something that gave me a rough idea in somewhat controlled conditions. Even something that just filtered the log to show the start/stop of download/encode phases. That would show when things were getting interleaved.

So I'd need to record a show on every channel to find all the mpeg4 channels. Maybe somebody in the Comcast or AVS forums knows about the mpeg4 status of my system.

**Comment by mackworth** [comment]
Well, remember the Tivo Suggestions, which in my case cover a lot of channels.  Also, you only need a minute or so on each of the other channels.

**Comment by mackworth** [comment]
Apparently, cTiVo crashes on 10.7; I have a fix, but I'll need a volunteer who's running 10.7 to try it out.

**Comment by tannebil** [comment]
I've got a 2011 MBA with Lion on it. What do you need tested?

**Comment by tannebil** [comment]
Check that. I upgraded it and can't find the Lion Installer to roll it back.

**Comment by Corbygue** [comment]
I upgraded to 2.5 because I was no longer able to download shows from one channel (Turner Classic Movies) and figured that I was dealing with the MPEG4 issue. Checking my Tivo shows that Comcast has indeed upgraded that channel. But my downloads are still failing after installing 2.5. The logs say "Show may be damaged/incomplete."

I haven't found problems with any other channels, and I can still download older recordings from TCM. And I can download the new programs that fail with cTivo onto my iPad using the Tivo app.

**Comment by mackworth** [comment]
Which "Format" are you using?  If `Decrypted TiVo`, then beta1 doesn't check that case. A workaround is to go to to Channels, and hit Test All Channels. Wait a while and it will confirm which channels are using the new MPEG4 format.

**Comment by Corbygue** [comment]
I use HandBrake AppleTV. Should that work?

**Comment by mackworth** [comment]
It should. Look in Preferences>Channels. Are any channels listed, especially TCM, and if so what do they say. If not, can you set Debug to Major; run a show and post the log (Help>Show Logs in Finder). You have to rename the .log file to .txt to attach to Github

**Comment by Corbygue** [comment]
Sorry for the delay, but of course my internet also went down while I was working on this....

Everything seems to be fine now for some reason. The h.264 issue was clearly my problem. I tried to download from some of the other channels that were listed as h.264 and had the same result of not being able to download.

But then I started getting a "server busy" notification from the Tivo that I'd never seen. It wouldn't go away until I rebooted the Tivo and my computer (which I'd already done a couple of times as I was trying to troubleshoot). Once I did that everything is working and I've been able to download a movie.

Thanks for you help.

**Comment by ooglek** [comment]
I'm having an issue with the beta as well. The reason I installed it was I was having trouble with subscriptions. I would add a subscription for "NBC Nightly News with Lester Holt" and on 2.4.2 and 2.4.4 it would download from my TiVo Bolt the current recorded episodes, but new ones would not automatically download, despite showing up in the list. 

I installed the beta, but now I cannot seem to get any of the shows to download. 

I tried the "Test All Channels" but only one channel showed up and all the rest of the tests "timed out" according to logs (attached). I've attached a few screenshots as well, but both are from 2.4.2 and 2.4.4, where subscriptions aren't working.

[com.cTiVo.cTiVo 2016-10-13 15-14.txt](https://github.com/dscottbuch/cTiVo/files/527564/com.cTiVo.cTiVo.2016-10-13.15-14.txt)
![ctivosnapz002](https://cloud.githubusercontent.com/assets/100755/19355831/c89deb5a-9139-11e6-843c-ea92ee065147.jpg)
![ctivosnapz001](https://cloud.githubusercontent.com/assets/100755/19355832/c8a3caac-9139-11e6-9204-b14d4cbf3a86.jpg)

**Comment by ooglek** [comment]
Whoops. I spoke too soon. I didn't realize that the downloading "waited" for a certain number of seconds before it started. With 2.4.x it started immediately, clearly I didn't wait long enough. 

It was going at 28MBps (224 Mbps) over Gigabit Ethernet, great download improvement. But then this logline hit:

`2016-10-13 11:45:18:981 MTDownload>connection:didReceiveData:@2418>self.urlBuffer length exceeded 50000000, switching to file based buffering`

And things slowed down a lot. Took a total of 4 minutes 37 seconds for the 30 minute HD show to download.

Eventually the download failed after the "Detecting Ads" phase (logs attached). 

I also noticed that when I turned off "Skip Commercials" the download ALWAYS fails.

Happy to help as much as possible with debugging.
[encode-fail.txt](https://github.com/dscottbuch/cTiVo/files/527627/encode-fail.txt)

**Comment by mackworth** [comment]
Yes, that is a change, the old one could overload the TiVo if you ran it twice in a row.  I also added the "Waiting" phase to communicate that. So, your initial logs wer completely bizarre, and I was going to recommend reinstallation.  This one looks much more normal. I do need a little more detail to diagnose. Can you set Preferences>Debug level to Major and run it again?

**Comment by ooglek** [comment]
Just installed the beta again (went back to 2.4.4 so I could download the video)

Attempted to manually start the download. Failed.

Logs attached. Download without Skip Commercials on still failing.

[encode-fail2.txt](https://github.com/dscottbuch/cTiVo/files/527721/encode-fail2.txt)

**Comment by mackworth** [comment]
Which version macOS are you on?

**Comment by mackworth** [comment]
Beta1 does not work properly on pre 10.11, which looks like your problem.  I'm about to release beta2 which should be fine.

**Comment by ooglek** [comment]
I'm running OSX 10.11.6 on an Early 2008 Mac Pro. Not sure that's the issue (I like to keep up to date).

**Comment by mackworth** [comment]
Hmm, that "completion code 4 and reason uncaught signal" looked like that.  Well, if you'd like to try the latest, not quite released, I'd love to know if this fixes it.....
https://dl.dropboxusercontent.com/u/21507587/cTiVo.zip

**Comment by ooglek** [comment]
Also, while we're discussing things, it drives me nuts that the same show gets downloaded with the same name regardless of the date it was recorded, like the nightly news. Any chance it could have the option to append the date, time, tivo ID or other unique information to the filename?

Tried the beta again this morning and no luck on downloading.

**Comment by ooglek** [comment]
I posted that with the old beta before I saw your post. 

Downloaded 663 build, same result. Output attached. One thought I had this morning was that I have ffmpegX installed on my Mac. While I would assume that wouldn't affect cTivo, since I'm sure you use absolute paths for external dependencies, I'm trying to think of things on my Mac that would potentially cause the failure that isn't due to cTivo.

Two attempts in the log, both failed. 

[encode-fail3.txt](https://github.com/dscottbuch/cTiVo/files/529731/encode-fail3.txt)

**Comment by ooglek** [comment]
Also, none of the betas are signed by Apple, so I have to undo security settings. Is that intentional?

**Comment by mackworth** [comment]
Hmm, can you set Debug to Detail and run again? Somehow it's running an executable that fails instantly.

On the file names, there is an option in Advanced Preference (Cmd-Opt-Comma) to name files however you like. See the Adv Pref help files for more details.

On the security settings, to be precise, do you mean changing from "Allow Apps downloaded from Mac App Store" to "... from Mac App Store and identified Developers"?  That is normal, and means that it is signed by me (as identified by Apple), but delivered outside the App Store. If it's requiring "Anywhere", that's a surprise.

**Comment by ooglek** [comment]
Looks like the issue may be with mencoder. I had created a custom Video Format for my iPhone 6 due to the changed resolution, and the included "iPhone" profile didn't get the quality/resolution I wanted. I tried the included "iPhone" one and it failed too. I tried with the "ffmpeg ComSkip/5.1" Video Format and that is progressing, I'll update when it finishes.

> max : /Applications/cTiVo.app/Contents/MacOS --> ./mencoder
> Illegal instruction
> Exit 132

Maybe this is due to my older CPUs? Or something else in my system? Or a linked library?

> mencoder:
> 	@executable_path/../lib/libiconv.2.dylib (compatibility version 8.0.0, current version 8.1.0)
> 	@executable_path/../lib/libncurses.6.dylib (compatibility version 6.0.0, current version 6.0.0)
> 	@executable_path/../lib/libgnutls.30.dylib (compatibility version 37.0.0, current version 37.6.0)
> 	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1258.1.0)
> 	/System/Library/Frameworks/VideoDecodeAcceleration.framework/Versions/A/VideoDecodeAcceleration (compatibility version 1.0.0, current version 1.0.0)
> 	/System/Library/Frameworks/QuartzCore.framework/Versions/A/QuartzCore (compatibility version 1.2.0, current version 1.11.0)
> 	@executable_path/../lib/libpng16.16.dylib (compatibility version 42.0.0, current version 42.0.0)
> 	@executable_path/../lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.8)
> 	@executable_path/../lib/libjpeg.9.dylib (compatibility version 11.0.0, current version 11.0.0)
> 	@executable_path/../lib/libbluray.1.dylib (compatibility version 11.0.0, current version 11.2.0)
> 	@executable_path/../lib/libbz2.1.0.dylib (compatibility version 1.0.0, current version 1.0.6)
> 	@executable_path/../lib/liblzo2.2.dylib (compatibility version 3.0.0, current version 3.0.0)
> 	@executable_path/../lib/libmad.0.dylib (compatibility version 3.0.0, current version 3.1.0)
> 	@executable_path/../lib/libvorbisenc.2.dylib (compatibility version 3.0.0, current version 3.11.0)
> 	@executable_path/../lib/libvorbis.0.dylib (compatibility version 5.0.0, current version 5.8.0)
> 	@executable_path/../lib/libtheoradec.1.dylib (compatibility version 3.0.0, current version 3.4.0)
> 	@executable_path/../lib/libogg.0.dylib (compatibility version 9.0.0, current version 9.2.0)
> 	@executable_path/../lib/libopus.0.dylib (compatibility version 6.0.0, current version 6.2.0)
> 	@executable_path/../lib/libdv.4.dylib (compatibility version 5.0.0, current version 5.3.0)
> 	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
> 	@executable_path/../lib/libtwolame.0.dylib (compatibility version 1.0.0, current version 1.0.0)
> 	@executable_path/../lib/libfaac.0.dylib (compatibility version 1.0.0, current version 1.0.0)
> 	@executable_path/../lib/libx264.148.dylib (compatibility version 0.0.0, current version 0.0.0)
> 	@executable_path/../lib/libmp3lame.0.dylib (compatibility version 1.0.0, current version 1.0.0)
> 	/System/Library/Frameworks/CoreVideo.framework/Versions/A/CoreVideo (compatibility version 1.2.0, current version 1.5.0)
> 

How can I increase the memory buffer? I've got 18GB and happy to let cTivo use more than 50MB if there isn't a performance issue doing so. 

`2016-10-14 14:06:39:387 MTDownload>connection:didReceiveData:@2438>self.urlBuffer length exceeded 50000000, switching to file based buffering`

I love that ffmpeg uses all of my CPU cores, that's awesome.

I've included the failures with the Video Format (you might want to include that in your logging) in the latest version.

Thanks for the tip about the advanced Preferences -- I didn't realize that was there. Huzzah for cTivo! :-)

**Comment by ooglek** [comment]
Oh, and I forgot -- both betas are not signed enough for OSX.

![system preferencessnapz001](https://cloud.githubusercontent.com/assets/100755/19400059/655350c8-9222-11e6-92e9-7fb6fcd290ba.jpg)
![findersnapz002](https://cloud.githubusercontent.com/assets/100755/19400057/6550b20a-9222-11e6-8fbc-00364dd3a3b2.jpg)
![findersnapz001](https://cloud.githubusercontent.com/assets/100755/19400058/6553058c-9222-11e6-8cc2-4cc9c099ff77.jpg)

**Comment by mackworth** [comment]
So, yes, after some further research, the issue is the older processor. I'm glad ffmpeg is running for you. Have you (or could you) run all the other embedded apps (e.g. try Handspring Apple TV, captions, and commercial marking)?

**Comment by mackworth** [comment]
The buffer is used to avoid writing the downloaded file to disk, instead feeding it directly into the tivodecode process (and hence into the other chains). The issue is that if you gets 50M behind, then the CPU is generally not keeping up with the transfer, so you'll get further and further behind with a larger and larger buffer, which if you set it too big will affect the performance of the rest of the system, including the encoders themselves.

We're going to deprecate mencoder in general. Nobody's working on it, and it's getting old and buggy.  I'd still like to have it work in this release though.

Not seeing the "the failures with the Video Format"; did you mean to attach a log?

I'm looking into the code signing issue.

**Comment by mackworth** [comment]
here's another beta.  Let me know if you get the signing error and if mencoder is still broken.

**Comment by mackworth** [comment]
I would still like to know if the other apps are breaking (from the previous beta)

**Comment by ooglek** [comment]
1. Signing works again! cTivo-10.7 did not throw any security errors once my security settings were set back to the App Store + Developers option.

2. ffmpeg comSkip works. 
  My iPhone mid-res profile does not. 
  iPhone Video Format did not work. 
  HandBrake AppleTV worked, though there are some interlacing artifacts on the non-HD source.   
  MP4 ffmpeg worked.

Looks like mencoder is still a problem. While on my Mac Pro mencoder actually starts to run, it looks like it is still dying further in with another Illegal Instruction.

> max : /Applications/cTiVo-10.7.app/Contents/MacOS --> cat ~/Downloads/cTivo/Robot\ Chicken-\ The\ Angelic\ Sounds\ of\ Mike\ Giggling.mp4 |  /Applications/cTiVo-10.7.app/Contents/MacOS/mencoder -ovc lavc -lavcopts vcodec=mpeg4:vbitrate=1500:autoaspect -oac mp3lame -lameopts br=128 -of lavf -lavfopts format=mov -hr-edl-seek -o '/Users/beckman/Downloads/cTivo/BBC World News--2016-09-31.avi' -
> MEncoder 1.3.0-4.2.1 (C) 2000-2016 MPlayer Team
> Reading from stdin...
> success: format: 0  data: 0x0 - 0x0
> libavformat version 57.25.100 (internal)
> libavformat file format detected.
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x107aeb7b0]stream 0, timescale not set
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x107aeb7b0]Protocol name not provided, cannot determine if input is local or a network protocol, buffers and access patterns cannot be configured optimally without knowing the protocol
> Illegal instruction
> Exit 132

Logs of all my tests today is attached (large)

[com.cTiVo.cTiVo 2016-10-15 15-20.txt](https://github.com/dscottbuch/cTiVo/files/531430/com.cTiVo.cTiVo.2016-10-15.15-20.txt)

**Comment by ooglek** [comment]
ffmpeg ComSkip/5.1 worked great, commercials were skipped well and correctly with my test video. I just have to go figure out how to cut out the 5.1 to stereo and drop the quality down (shooting for 200-300MB for a 22 minute show (after commercials are cut)).

**Comment by mackworth** [comment]
Great. 
Have you tried captions? 
I take it 2.4.4 worked fine with mencoder for you?
I think for the 10.7 compatible ctivo, I may have to roll back to that version. 
I'll add an option to not do the AC3 to the script.

**Comment by ooglek** [comment]
2.4.4 with mencoder yes, that works for me, I've gone back to 2.4.4 a few times in the last few days to get the file off the TiVo and onto my phone, using my mencoder Video Format.

When you say 10.7 compatible cTivo, what do you mean? I'm running 10.11, but I thought the mencoder issue was due to my CPU family, not OS version. It might be a term of art that I don't understand.

Captions -- you mean subtitles? I've enabled that for this run, and the SRT file was created and works in VLC with the ffmpeg ComSkip Video Format.

Is there any way to change the video quality in the ffmpeg ComSkip format? Right now a 30 minute HD show is, after commercials are removed down to 26 minutes, taking about 1.05GB. My goal is around 200-300MB. How would I drop the quality? I know it is somewhere in that bash script... :-)

**Comment by mackworth** [comment]
There are too many problems to continue supporting 10.7 and 10.8, but it's critical that I provide a version that handles the MP4 transition, so I've created two versions, one which only runs in 10.9 and above and one that runs in the older ones, but uses the older executables. 

In particular, I'm giving up on mencoder.  If you agree with that, then you should be able to run the 10.9 version for everything else; if not, I'm putting the old version in the 10.7 version. So you can run that, and everything should be fine as it was, but maybe a little slower and a little less compatible vs the new mencoder.

So, if you could do one more set of tests for me, I'd appreciate it. You have an hard-to-recreate combination of very old hardware and modern OS.

First, run [the new 10.9+ version here](https://dl.dropboxusercontent.com/u/21507587/cTiVo.zip):  and run Iphone or other mencoder Format, and confirm that you now get a better error msg.  Second, run [this 10.7 version here](https://dl.dropboxusercontent.com/u/21507587/cTiVo-10.7.zip), and try mencoder again to confirm it works. Third, if you could run both versions (one at a time, please) and make sure that all the other executables work (comskip, subtitles, ffmpeg, Handbrake) in both versions; that would be excellent. This is my last step before releasing this beta.

Thanks for all your help on this.

ps In a bit, I'll generate a "Small ffmpeg comskip" version, which would both cap the resolution and reduce the quality (Part of what I need to do for the next beta). (You can do it yourself by creating a new Format, calling the script, but it requires figuring out the right parameters for ffmpeg.)

**Comment by ooglek** [comment]
10.9 iPhone Format: fails after marking commercials (expected). Still get an Illegal Instruction error with mencoder. Failure causes cTivo to try again, and downloads the file again, detects commercials again, etc. I didn't wait to have it try 3 times. There weren't any helpful error messages beyond "Failed" and a growl message saying it was not going to try again. Also, I tried downloading a second time after deleting the first, and it just failed quickly three times in a row. 

> 2016-10-16 13:22:43:887 MTDownload>download@1750>Starting URL http://dvr-2d91.local.:80/download/BBC%20World%20News.TiVo?Container=%2FNowPlaying&id=515 for show BBC World News in 0.0 seconds
> 2016-10-16 13:22:44:021 MTDownload>observeValueForKeyPath:ofObject:change:context:@165>Changing DL status of BBC World News (Living Room) to Downloading (1)
> 2016-10-16 13:22:44:184 MTDownload>connection:willSendRequestForAuthenticationChallenge:@2466>Using proposed Credential for Living Room
> 2016-10-16 13:22:44:370 MTTask>completeProcess@172>Finished task encode of show BBC World News with completion code 4 and reason uncaught signal
> 2016-10-16 13:22:44:370 MTTask>failedTaskCompletion@188>Task encode failed
> 2016-10-16 13:22:44:371 MTDownload>rescheduleShowWithDecrementRetries:@2055>BBC World News (Living Room) downloaded 0K of 348971 KB; 0.0% processed
> 2016-10-16 13:22:44:371 MTDownload>cancel@2116>Canceling of BBC World News
> 2016-10-16 13:22:44:371 MTTaskChain>cancel@170>Canceling task encode for show BBC World News
> 2016-10-16 13:22:44:371 MTTask>cancel@102>Terminating task encode
> 2016-10-16 13:22:44:371 MTTaskChain>cancel@170>Canceling task decrypt for show BBC World News
> 2016-10-16 13:22:44:371 MTTask>cancel@102>Terminating task decrypt
> 2016-10-16 13:22:44:371 MTDownload>cancel@2140>Waiting 0.000002 seconds for write data to complete during cancel
> 2016-10-16 13:22:44:371 MTDownload>cleanupFiles@463>BBC World News cleaningup files
> 2016-10-16 13:22:44:372 MTDownload>rescheduleShowWithDecrementRetries:@2068>Stopping at Downloading, restarting download of BBC World News with progress at 0.000000 with previous check at 2016-10-16 17:22:43 +0000
> 2016-10-16 13:22:44:372 MTTiVoManager>notifyForDownload:withTitle:subTitle:isSticky:forNotification:@1403>Notify: TiVo show failed/nRetrying: Can't Download
> 2016-10-16 13:22:44:373 MTDownload>rescheduleShowWithDecrementRetries:@2098>Decrementing retries to 2

2.5.0 Beta 2 (665) on Mac Pro Early 2008 using OSX 10.11.6:
1. ✅  iPhone Format worked (mencoder), no skip or subtitles
2. ✅  My own iPhone medium-res Format (mencoder) with commercial skipping, which removed the commercials successfully.
3. ⚠️  ffmpeg ComSkip/5.1 with skip and subtitles -- the video and audio and commercial skipping were fine, but the .srt file came out to be 3 bytes and VLC failed to load it. This may be expected if there are no subtitles in a feed, but I leave that to you. 
4. ⚠️  Used ffmpeg ComSkip/5.1 with skip and subtitles. Subtitles worked this time. Though Got a file from the TiVo that warned: "Notify: Warning: Show may be damaged/incomplete. Transfer is too short: Possible Problem" 565563.1 of 650117.1 KB expected; 87.0%. Process continued, completed. Watched it, seems to have worked correctly, nothing but commercials removed from the video. What's the reason for the file size measurement/warnings?
5. ✅ Used Handbrake iPhone. Knocked it out fast, worked fine. No commercial skip (not available), didn't try subtitles.

I'll try and do a more comprehensive test on the 10.9+ version later.

**Comment by mackworth** [comment]
Good stuff; thank you. 

So you've gotten two different error msgs for the older processor. I've added the second test for [this 10.9+ version here](https://dl.dropboxusercontent.com/u/21507587/cTiVo.zip), if you could try the iphone version once more to see if my better error msg comes up (and if the shorter file message is still there, see below).

Yes, ccextractor leaves a 3-character file when there's no subtitles. (The three chars are how every UTF8 should start). But no reason to hang onto those, so I've now deleted them.

The reason for the warning is that sometimes the TiVo will send short files, just because it's got a digital glitch in the file. Annoying, and so we warn when that's the case. However, the actual files sent are usually slightly shorter than we're told they are when we start transmission. I've used 90% in the past as a dividing line between "too short, must be a problem" and "a little short, probably ok", but I've moved it down to 85% based on your evidence here..

**Comment by mackworth** [comment]
Closing and moving to Beta 2 release discussion #206

**Comment by mackworth** [comment]
@ooglek  If you're still running 10.7, could you do Check For Updates in the cTiVo menu and let me know if it works?

**Comment by ooglek** [comment]
I was never running 10.7, just had an older Early 2008 Mac Pro, running 10.11. 

I just tested it and the Check for Updates from 2.5.0 beta 2 saw the update. I haven't run the update yet though.

**Comment by mackworth** [comment]
Ok, thanks for checking!

**Comment by tannebil** [comment]
Sorry but the MBA that was running 10.7 is now running 10.8 and is taking a vacation in Argentina without me

**Comment by mackworth** [comment]
Oh, that's so sad!  Can't you catch up to it?

And actually, now that you mention it and pushing my luck...if you left ctivo on it and it has an internet connection and you're in touch with the peripatetic owner,... could they just run it, do the Check for Updates (in the cTiVo menu) and confirm that the resulting screen doesn't say "you have the latest version" and that it does says "This version is only for systems using 10.7 or 10.8."?    No actual TiVo work required.

**Comment by tannebil** [comment]
I do have remote access to it although it doesn’t have cTiVo on it at the moment. Can I install it and do the test or will it get stuck looking for a TiVo?

**Comment by mackworth** [comment]
Sure. I haven't pulled down the 2.4.4 version, so you can get it here:
https://github.com/dscottbuch/cTiVo/releases/download/2.4.4/cTiVo.zip
Then run it, and you should be immediately able to cTiVo>Check For Updates.  No, it will complain it can't find one, but that shouldn't interfere with the menus.

**Comment by tannebil** [comment]
I’m not going to be able to do it until I return to Argentina at the end of February

**Comment by mackworth** [comment]
Don't worry 'bout it. I'm seeing updates to the 10-7 version and no fatal crashes, so it's going ok.
